domain/access/modelmigration/import_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmigration
+
+import (
+	"context"
+	"time"
+
+	"github.com/juju/description/v8"
+	"github.com/juju/names/v5"
+	jc "github.com/juju/testing/checkers"
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	coremodel "github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/juju/core/user"
+	loggertesting "github.com/juju/juju/internal/logger/testing"
+)
+
+type importSuite struct {
+	coordinator *MockCoordinator
+	service     *MockImportService
+}
+
+var _ = gc.Suite(&importSuite{})
+
+func (s *importSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+
+	s.coordinator = NewMockCoordinator(ctrl)
+	s.service = NewMockImportService(ctrl)
+
+	return ctrl
+}
+
+func (s *importSuite) newImportOperation() *importOperation {
+	return &importOperation{
+		service: s.service,
+	}
+}
+
+func (s *importSuite) TestRegisterImport(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.coordinator.EXPECT().Add(gomock.Any())
+
+	RegisterImport(s.coordinator, loggertesting.WrapCheckLog(c))
+}
+
+func (s *importSuite) TestNoModelUserPermissions(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	// Empty model.
+	model := description.NewModel(description.ModelArgs{})
+
+	op := s.newImportOperation()
+	err := op.Execute(context.Background(), model)
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *importSuite) TestImport(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	model := description.NewModel(description.ModelArgs{})
+	modelUUID := model.Tag().Id()
+	modelID := permission.ID{
+		ObjectType: permission.Model,
+		Key:        modelUUID,
+	}
+	steveTag := names.NewUserTag("steve")
```

Comment:
**nitpick:** Make the test boring. `steveTag` is the creatorTag for both user (in both tests). It would be so much simpler to name it `creator` to avoid reviewer to wonder who is steve? (even it make me think about memes about stranger things, which make me smile).

Reply:
Good point, changed